io/net/ethtool/ethtool_tests.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash -e
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+
+# Tests the network driver and interface with 'ethtool' command.
+# Different parameters are specified in Parameters section of multiplexer file.
+# Interfaces are specified in Interfaces section of multiplexer file.
+# This test needs to be run as root.
+
+PATH=$(avocado "exec-path"):$PATH
+
+# Install dependencies
+python << END
+from avocado.utils.software_manager import SoftwareManager;
+if SoftwareManager().check_installed("ethtool") is False:
+    print "Ethtool is not installed"
+    if SoftwareManager().install("ethtool") is False:
+        print "Not able to install ethtool"
+        exit(1)
+END
+exit
+
+[[ -z $interface ]] && interface="eth1"
+echo "Interface: $interface"
+
+ip link show dev $interface > /dev/null 2>&1 || {
+    avocado_warn "$interface does not exist"
+    exit 1
+}
+
+echo "Parameter: ${arg}"
+
+ifup $interface
+for command in ifdown ifup; do
+    sleep 10
+    $command $interface
+    sleep 10
+    echo "Ethool, $command:"
+    output_file="$AVOCADO_TEST_OUTPUTDIR/output_$command"
+    ethtool ${arg} ${interface} $action_elapse | tee "$output_file"
```

Comment:
This is not redirecting the stderr to the `$output_file`. In consequence, the the `elif` in line 57 will not find the *Operation not supported* message (the same applies to the other `elif`s).